packages/smooth_app/test/cache/cache_test.dart

```diff
@@ -0,0 +1,45 @@
+import 'dart:typed_data';
+
+import 'package:flutter_test/flutter_test.dart';
+import 'package:path_provider_platform_interface/path_provider_platform_interface.dart';
+import 'package:smooth_app/cache/files/files_cache.dart';
+
+import '../utils/path_provider_mock.dart';
+
+void main() {
+  TestWidgetsFlutterBinding.ensureInitialized();
+  PathProviderPlatform.instance = MockedPathProviderPlatform();
+
+  const String testFileContent = 'Hello World';
+  const String fileKey = 'hello';
```

Comment:
In the real world, what is going to be the key?

Reply:
It can be the name of product + type. Eg: `123684894_ingredients.jpg`

Reply:
My assertion was that the cache should be "content" agnostic.
However, a "Product cache" should indeed compute that key.

What would you think about this?

Reply:
You can be content-agnostic as long as you're lucky enough not to have colliding keys between several systems.
Anyway, some day you'll need to implement the method that computes the key for a (Product + ingredient).
You're right, that can be part of another PR.